src/server/project.ts

```diff
@@ -1730,6 +1731,17 @@ namespace ts.server {
                 this.projectService.openFiles,
                 (_, fileName) => this.projectService.tryGetDefaultProjectForFile(toNormalizedPath(fileName)) === this);
         }
+
+        /*@internal*/
+        watchNodeModulesDirectory(directoryPath: string, cb: DirectoryWatcherCallback) {
+            return this.projectService.watchFactory.watchDirectory(

```

Comment:
@sheetalkamat, does this handle sharing watchers for the same directory under the hood, or is there a different utility I can use for that? I noticed when testing that there’s often already a watch on node_modules for closed script info from the AutoImportProviderProject, since its root names are all in node_modules.

Reply:
This does not share the watches at all between projects... 
You would want to look for `updateSharedExtendedConfigFileWatcher` which does some sharing on extended config file watching ...  You may also look at `createConfigFileWatcherForParsedConfig` which is more involved though since it has to do with managing projects, parsedconfigs and open files so that might not be what you need but might help if you want to share watches between closed script infos as well as this one..